File: mobile/customer-app/app/src/main/java/com/sukashawarma/customer/ui/components/HomeHeader.kt
package com.sukashawarma.customer.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicTextField
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.KeyboardArrowDown
import androidx.compose.material.icons.filled.LocalFireDepartment
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.Search
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.sukashawarma.customer.ui.theme.LilitaOne
import com.sukashawarma.customer.ui.theme.SukaBorder
import com.sukashawarma.customer.ui.theme.SukaBrown
import com.sukashawarma.customer.ui.theme.SukaGreen
import com.sukashawarma.customer.ui.theme.SukaInk
import com.sukashawarma.customer.ui.theme.SukaMuted
import com.sukashawarma.customer.ui.theme.SukaOrange
import com.sukashawarma.customer.ui.theme.SukaTint

/**
 * Gradient latar belakang khas SukaBrown untuk seluruh Header Brand.
 */
private val SukaBrandHeaderGradient = Brush.verticalGradient(
    listOf(
        Color(0xFF4A0E03),
        SukaBrown,
        Color(0xFF5E1203)
    )
)

/**
 * 1. Header Beranda Brand Heritage (HomeBrandHeader).
 * Ramping, elegan, dan proporsional (~118dp) agar tidak memotong kartu Best Seller di bawahnya.
 */
@Composable
fun HomeBrandHeader(
    namaOutlet: String,
    buka: Boolean,
    onGantiOutlet: () -> Unit,
    onBukaProfil: () -> Unit,
    onBukaNotifikasi: () -> Unit = {},
    unreadCount: Int = 0,
    modifier: Modifier = Modifier
) {
    Surface(
        modifier = modifier.fillMaxWidth(),
        color = SukaBrown,
        shape = RoundedCornerShape(bottomStart = 22.dp, bottomEnd = 22.dp),
        shadowElevation = 5.dp
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(SukaBrandHeaderGradient)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .statusBarsPadding()
                    .padding(horizontal = 14.dp, vertical = 8.dp),
                verticalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                // Baris 1: Logo Brand, Nama Brand & Aksi Kanan (Notif & Avatar)
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Row(
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        Box(
                            modifier = Modifier
                                .size(32.dp)
                                .clip(RoundedCornerShape(9.dp))
                                .background(SukaOrange),
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Filled.LocalFireDepartment,
                                contentDescription = null,
                                tint = SukaInk,
                                modifier = Modifier.size(20.dp)
                            )
                        }

                        Column {
                            Text(
                                text = "SUKA SHAWARMA",
                                fontFamily = LilitaOne,
                                fontSize = 15.sp,
                                color = Color(0xFFFFF4EB),
                                letterSpacing = 0.5.sp
                            )
                            Text(
                                text = "Otentik • Panggang • Gurih",
                                style = MaterialTheme.typography.labelSmall.copy(
                                    color = SukaOrange,
                                    fontSize = 9.sp,
                                    fontWeight = FontWeight.Bold
                                )
                            )
                        }
                    }

                    // Aksi Kanan: Lonceng Notifikasi & Avatar Profil
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(7.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Box(
                            modifier = Modifier
                                .size(32.dp)
                                .clip(CircleShape)
                                .background(Color.White.copy(alpha = 0.12f))
                                .border(BorderStroke(1.dp, Color.White.copy(alpha = 0.2f)), CircleShape)
                                .bounceClick(scaleDown = 0.92f) { onBukaNotifikasi() },
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Filled.Notifications,
                                contentDescription = "Notifikasi",
                                tint = Color(0xFFFFF4EB),
                                modifier = Modifier.size(16.dp)
                            )
                            if (unreadCount > 0) {
                                Box(
                                    modifier = Modifier
                                        .align(Alignment.TopEnd)
                                        .padding(top = 5.dp, end = 5.dp)
                                        .size(6.dp)
                                        .clip(CircleShape)
                                        .background(SukaOrange)
                                )
                            }
                        }

                        Box(
                            modifier = Modifier
                                .size(32.dp)
                                .clip(CircleShape)
                                .background(SukaOrange)
                                .bounceClick(scaleDown = 0.94f) { onBukaProfil() },
                            contentAlignment = Alignment.Center
                        ) {
                            Text(
                                text = "SK",
                                style = MaterialTheme.typography.labelMedium.copy(
                                    fontWeight = FontWeight.ExtraBold,
                                    color = SukaInk,
                                    fontSize = 11.sp
                                )
                            )
                        }
                    }
                }

                // Baris 2: Kartu Outlet Ramping & Elegan (Floating Card)
                Surface(
                    modifier = Modifier
                        .fillMaxWidth()
                        .bounceClick(scaleDown = 0.98f) { onGantiOutlet() },
                    shape = RoundedCornerShape(14.dp),
                    color = Color.White,
                    shadowElevation = 3.dp,
                    border = BorderStroke(1.dp, SukaBorder)
                ) {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = 10.dp, vertical = 7.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Row(
                            verticalAlignment = Alignment.CenterVertically,
                            horizontalArrangement = Arrangement.spacedBy(8.dp),
                            modifier = Modifier.weight(1f)
                        ) {
                            Box(
                                modifier = Modifier
                                    .size(28.dp)
                                    .clip(CircleShape)
                                    .background(SukaOrange.copy(alpha = 0.15f)),
                                contentAlignment = Alignment.Center
                            ) {
                                Icon(
                                    imageVector = Icons.Filled.LocationOn,
                                    contentDescription = null,
                                    tint = SukaOrange,
                                    modifier = Modifier.size(16.dp)
                                )
                            }

                            Column(verticalArrangement = Arrangement.spacedBy(1.dp)) {
                                Row(
                                    verticalAlignment = Alignment.CenterVertically,
                                    horizontalArrangement = Arrangement.spacedBy(5.dp)
                                ) {
                                    Text(
                                        text = namaOutlet,
                                        style = MaterialTheme.typography.titleSmall.copy(
                                            fontWeight = FontWeight.ExtraBold,
                                            color = SukaInk,
                                            fontSize = 13.sp
                                        ),
                                        maxLines = 1,
                                        overflow = TextOverflow.Ellipsis
                                    )
                                    Box(
                                        modifier = Modifier
                                            .size(4.dp)
                                            .clip(CircleShape)
                                            .background(if (buka) SukaGreen else SukaMuted)
                                    )
                                    Text(
                                        text = if (buka) "Buka" else "Tutup",
                                        style = MaterialTheme.typography.labelSmall.copy(
                                            fontSize = 9.sp,
                                            fontWeight = FontWeight.Bold,
                                            color = if (buka) SukaGreen else SukaMuted
                                        )
                                    )
                                }
                                Text(
                                    text = if (buka) "Siap saji dalam 15–20 menit" else "Tidak menerima pesanan",
                                    style = MaterialTheme.typography.bodySmall.copy(
                                        fontSize = 10.sp,
                                        color = SukaMuted
                                    )
                                )
                            }
                        }

                        // Tombol Pill "Ganti ▾"
                        Box(
                            modifier = Modifier
                                .clip(RoundedCornerShape(8.dp))
                                .background(SukaTint)
                                .border(BorderStroke(1.dp, SukaOrange.copy(alpha = 0.3f)), RoundedCornerShape(8.dp))
                                .padding(horizontal = 8.dp, vertical = 4.dp)
                        ) {
                            Row(
                                verticalAlignment = Alignment.CenterVertically,
                                horizontalArrangement = Arrangement.spacedBy(2.dp)
                            ) {
                                Text(
                                    text = "Ganti",
                                    style = MaterialTheme.typography.labelSmall.copy(
                                        fontWeight = FontWeight.ExtraBold,
                                        color = SukaBrown,
                                        fontSize = 11.sp
                                    )
                                )
                                Icon(
                                    imageVector = Icons.Filled.KeyboardArrowDown,
                                    contentDescription = "Ganti Outlet",
                                    tint = SukaBrown,
                                    modifier = Modifier.size(14.dp)
                                )
                            }
                        }
                    }
                }
            }
        }
    }
}

/**
 * 2. Header Brand untuk Halaman Menu (MenuBrandHeader).
 * Latar SukaBrown melengkung, selector outlet ringkas, serta kolom pencarian terintegrasi.
 */
@Composable
fun MenuBrandHeader(
    namaOutlet: String,
    buka: Boolean,
    kueriPencarian: String,
    onUbahKueri: (String) -> Unit,
    onGantiOutlet: () -> Unit,
    onBukaProfil: () -> Unit,
    modifier: Modifier = Modifier,
    kategoriContent: (@Composable () -> Unit)? = null
) {
    Surface(
        modifier = modifier.fillMaxWidth(),
        color = SukaBrown,
        shape = RoundedCornerShape(bottomStart = 22.dp, bottomEnd = 22.dp),
        shadowElevation = 5.dp
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(SukaBrandHeaderGradient)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .statusBarsPadding()
                    .padding(top = 8.dp, bottom = if (kategoriContent != null) 10.dp else 12.dp),
                verticalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                // Baris Atas: Judul Menu, Outlet Pill & Avatar
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = 14.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Row(
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        Box(
                            modifier = Modifier
                                .size(32.dp)
                                .clip(RoundedCornerShape(9.dp))
                                .background(SukaOrange),
                            contentAlignment = Alignment.Center
                        ) {
                            Icon(
                                imageVector = Icons.Filled.LocalFireDepartment,
                                contentDescription = null,
                                tint = SukaInk,
                                modifier = Modifier.size(20.dp)
                            )
                        }

                        Column {
                            Text(
                                text = "MENU & KATALOG",
                                fontFamily = LilitaOne,
                                fontSize = 15.sp,
                                color = Color(0xFFFFF4EB),
                                letterSpacing = 0.5.sp
                            )
                            Text(
                                text = "Pilihan Shawarma Otentik",
                                style = MaterialTheme.typography.labelSmall.copy(
                                    color = SukaOrange,
                                    fontSize = 9.sp,
                                    fontWeight = FontWeight.Bold
                                )
                            )
                        }
                    }

                    // Aksi Kanan: Outlet Selector Chip & Profil
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(6.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        // Pill Outlet
                        Surface(
                            shape = RoundedCornerShape(10.dp),
                            color = Color.White.copy(alpha = 0.15f),
                            border = BorderStroke(1.dp, Color.White.copy(alpha = 0.25f)),
                            modifier = Modifier
                                .bounceClick(scaleDown = 0.96f) { onGantiOutlet() }
                        ) {
                            Row(
                                modifier = Modifier.padding(horizontal = 8.dp, vertical = 5.dp),
                                verticalAlignment = Alignment.CenterVertically,
                                horizontalArrangement = Arrangement.spacedBy(4.dp)
                            ) {
                                Icon(
                                    Icons.Filled.LocationOn,
                                    contentDescription = null,
                                    tint = SukaOrange,
                                    modifier = Modifier.size(13.dp)
                                )
                                Text(
                                    text = namaOutlet,
                                    color = Color.White,
                                    fontSize = 11.sp,
                                    fontWeight = FontWeight.Bold,
                                    maxLines = 1,
                                    overflow = TextOverflow.Ellipsis
                                )
                                Icon(
                                    Icons.Filled.KeyboardArrowDown,
                                    contentDescription = null,
                                    tint = Color.White,
                                    modifier = Modifier.size(13.dp)
                                )
                            }
                        }

                        // Avatar
                        Box(
                            modifier = Modifier
                                .size(32.dp)
                                .clip(CircleShape)
                                .background(SukaOrange)
                                .bounceClick(scaleDown = 0.94f) { onBukaProfil() },
                            contentAlignment = Alignment.Center
                        ) {
                            Text(
                                text = "SK",
                                style = MaterialTheme.typography.labelMedium.copy(
                                    fontWeight = FontWeight.ExtraBold,
                                    color = SukaInk,
                                    fontSize = 11.sp
                                )
                            )
                        }
                    }
                }

                // Baris Tengah: Search Box Terintegrasi di Header
                Surface(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = 14.dp)
                        .height(42.dp),
                    shape = RoundedCornerShape(12.dp),
                    color = Color.White,
                    shadowElevation = 2.dp,
                    border = BorderStroke(1.dp, SukaBorder)
                ) {
                    Row(
                        modifier = Modifier
                            .fillMaxSize()
                            .padding(horizontal = 12.dp),
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Filled.Search,
                            contentDescription = "Cari",
                            tint = SukaMuted,
                            modifier = Modifier.size(18.dp)
                        )

                        BasicTextField(
                            value = kueriPencarian,
                            onValueChange = onUbahKueri,
                            modifier = Modifier.weight(1f),
                            singleLine = true,
                            textStyle = TextStyle(
                                fontSize = 13.sp,
                                color = SukaInk,
                                fontWeight = FontWeight.Medium
                            ),
                            decorationBox = { innerTextField ->
                                if (kueriPencarian.isEmpty()) {
                                    Text(
                                        text = "Cari menu favorit, shawarma, saus...",
                                        color = SukaMuted,
                                        fontSize = 12.sp
                                    )
                                }
                                innerTextField()
                            }
                        )

                        if (kueriPencarian.isNotEmpty()) {
                            Icon(
                                imageVector = Icons.Filled.Close,
                                contentDescription = "Hapus",
                                tint = SukaMuted,
                                modifier = Modifier
                                    .size(18.dp)
                                    .clickable { onUbahKueri("") }
                            )
                        }
                    }
                }

                // Baris Bawah: Kategori Terintegrasi (Menyatu dalam Header)
                if (kategoriContent != null) {
                    kategoriContent()
                }
            }
        }
    }
}

/**
 * 3. Header Brand Universal untuk Halaman Riwayat Pesanan & Profil (PageBrandHeader).
 * Memberikan konsistensi warna SukaBrown melengkung, tipografi LilitaOne, dan tombol navigasi/aksi.
 */
@Composable
fun PageBrandHeader(
    judul: String,
    subjudul: String? = null,
    onKembali: (() -> Unit)? = null,
    aksiKanan: (@Composable () -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    Surface(
        modifier = modifier.fillMaxWidth(),
        color = SukaBrown,
        shape = RoundedCornerShape(bottomStart = 22.dp, bottomEnd = 22.dp),
        shadowElevation = 5.dp
    ) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(SukaBrandHeaderGradient)
        ) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .statusBarsPadding()
                    .padding(horizontal = 16.dp, vertical = 12.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                // Sisi Kiri: Tombol Kembali ATAU Logo Api Brand
                if (onKembali != null) {
                    Box(
                        modifier = Modifier
                            .size(34.dp)
                            .clip(CircleShape)
                            .background(Color.White.copy(alpha = 0.15f))
                            .border(BorderStroke(1.dp, Color.White.copy(alpha = 0.25f)), CircleShape)
                            .bounceClick(onClick = onKembali),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = "Kembali",
                            tint = Color(0xFFFFF4EB),
                            modifier = Modifier.size(18.dp)
                        )
                    }
                } else {
                    Box(
                        modifier = Modifier
                            .size(34.dp)
                            .clip(RoundedCornerShape(10.dp))
                            .background(SukaOrange),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.Filled.LocalFireDepartment,
                            contentDescription = null,
                            tint = SukaInk,
                            modifier = Modifier.size(20.dp)
                        )
                    }
                }

                // Sisi Tengah: Judul Halaman & Subjudul
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    modifier = Modifier
                        .weight(1f)
                        .padding(horizontal = 8.dp)
                ) {
                    Text(
                        text = judul,
                        fontFamily = LilitaOne,
                        fontSize = 17.sp,
                        color = Color(0xFFFFF4EB),
                        letterSpacing = 0.5.sp,
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis
                    )
                    if (subjudul != null) {
                        Text(
                            text = subjudul,
                            style = MaterialTheme.typography.bodySmall.copy(
                                color = SukaOrange,
                                fontSize = 10.sp,
                                fontWeight = FontWeight.SemiBold
                            ),
                            maxLines = 1,
                            overflow = TextOverflow.Ellipsis
                        )
                    }
                }

                // Sisi Kanan: Aksi Kustom ATAU Spacer Penyeimbang
                if (aksiKanan != null) {
                    aksiKanan()
                } else {
                    Spacer(modifier = Modifier.size(34.dp))
                }
            }
        }
    }
}

File: mobile/customer-ios-app/SukaShawarma/UI/Components/HomeHeader.swift
import SwiftUI

/// Padanan `ui/components/HomeHeader.kt`: tiga kepala cokelat melengkung.

private let bentukKepala = UnevenRoundedRectangle(bottomLeadingRadius: 22, bottomTrailingRadius: 22)

private struct LatarKepala: ViewModifier {
    func body(content: Content) -> some View {
        content
            .background {
                bentukKepala.fill(SukaGradien.kepala)
                    .bayangan(5)
                    .ignoresSafeArea(edges: .top)
            }
    }
}

private struct LogoApi: View {
    var ukuran: CGFloat = 32
    var body: some View {
        Image(systemName: "flame.fill")
            .font(.system(size: ukuran * 0.55))
            .foregroundStyle(Color.sukaInk)
            .frame(width: ukuran, height: ukuran)
            .background(Color.sukaOrange, in: RoundedRectangle(cornerRadius: ukuran * 0.28))
    }
}

private struct JudulMerek: View {
    let judul: String
    let sub: String
    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Text(judul).font(SukaFont.lilita(15)).tracking(0.5).foregroundStyle(Color.sukaCreamTeks)
            Text(sub).font(SukaFont.jakarta(11, weight: .bold)).foregroundStyle(Color.sukaOrange)
        }
    }
}

/// Avatar "SK" (area sentuh 48pt, lingkaran 36pt).
struct AvatarProfil: View {
    let aksi: () -> Void
    var body: some View {
        Button(action: aksi) {
            Text("SK")
                .font(SukaFont.jakarta(12, weight: .bold))
                .foregroundStyle(Color.sukaInk)
                .frame(width: 36, height: 36)
                .background(Color.sukaOrange, in: Circle())
                .frame(width: 48, height: 48)
        }
        .buttonStyle(.mentul(0.94))
        .accessibilityLabel("Profil")
    }
}

/// 1. Kepala Beranda: logo, lonceng, avatar, dan kartu outlet.
struct HomeBrandHeader: View {
    let namaOutlet: String
    let buka: Bool
    var unreadCount = 0
    let onGantiOutlet: () -> Void
    let onBukaProfil: () -> Void
    let onBukaNotifikasi: () -> Void

    var body: some View {
        VStack(spacing: 8) {
            HStack {
                HStack(spacing: 8) {
                    LogoApi()
                    JudulMerek(judul: "SUKA SHAWARMA", sub: "Otentik • Panggang • Gurih")
                }
                Spacer()
                Button(action: onBukaNotifikasi) {
                    Image(systemName: "bell.fill")
                        .font(.system(size: 14))
                        .foregroundStyle(Color.sukaCreamTeks)
                        .frame(width: 36, height: 36)
                        .background(Color.white.opacity(0.12), in: Circle())
                        .overlay(Circle().stroke(Color.white.opacity(0.2), lineWidth: 1))
                        .overlay(alignment: .topTrailing) {
                            if unreadCount > 0 {
                                Circle().fill(Color.sukaOrange).frame(width: 6, height: 6).padding(5)
                            }
                        }
                        .frame(width: 48, height: 48)
                }
                .buttonStyle(.mentul(0.92))
                .accessibilityLabel(unreadCount > 0 ? "Notifikasi, \(unreadCount) belum dibaca" : "Notifikasi")
                .accessibilityIdentifier("tombol-notifikasi")
                AvatarProfil(aksi: onBukaProfil)
            }

            Button(action: onGantiOutlet) {
                HStack {
                    HStack(spacing: 8) {
                        Image(systemName: "mappin")
                            .font(.system(size: 14, weight: .bold))
                            .foregroundStyle(Color.sukaOrange)
                            .frame(width: 28, height: 28)
                            .background(Color.sukaOrange.opacity(0.15), in: Circle())
                        VStack(alignment: .leading, spacing: 1) {
                            HStack(spacing: 4) {
                                Text(namaOutlet)
                                    .font(SukaFont.jakarta(13, weight: .bold))
                                    .foregroundStyle(Color.sukaInk)
                                    .lineLimit(1)
                                Circle().fill(buka ? Color.sukaGreen : Color.sukaMuted).frame(width: 4, height: 4)
                                Text(buka ? "Buka" : "Tutup")
                                    .font(SukaFont.jakarta(11, weight: .bold))
                                    .foregroundStyle(buka ? Color.sukaGreen : Color.sukaMuted)
                            }
                            Text(buka ? "Siap saji dalam 15–20 menit" : "Tidak menerima pesanan")
                                .font(SukaFont.jakarta(11)).foregroundStyle(Color.sukaMuted)
                        }
                    }
                    Spacer(minLength: 8)
                    HStack(spacing: 2) {
                        Text("Ganti").font(SukaFont.jakarta(11, weight: .bold))
                        Image(systemName: "chevron.down").font(.system(size: 10, weight: .bold))
                    }
                    .foregroundStyle(Color.sukaBrown)
                    .padding(.horizontal, 8).padding(.vertical, 4)
                    .background(Color.sukaTint, in: RoundedRectangle(cornerRadius: 8))
                    .overlay(RoundedRectangle(cornerRadius: 8).stroke(Color.sukaOrange.opacity(0.3), lineWidth: 1))
                }
                .padding(.horizontal, 12).padding(.vertical, 8)
                .kartuSuka(sudut: 14, elevasi: 3)
            }
            .buttonStyle(.mentul(0.98))
            .accessibilityHint("Ganti outlet")
            .accessibilityIdentifier("ganti-outlet")
        }
        .padding(.horizontal, 16)
        .padding(.top, 8).padding(.bottom, 12)
        .dynamicTypeSize(...DynamicTypeSize.xxxLarge)
        .modifier(LatarKepala())
    }
}

/// 2. Kepala tab Menu: pil outlet, avatar, kolom cari, dan (opsional) chip kategori.
struct MenuBrandHeader<Kategori: View>: View {
    let namaOutlet: String
    @Binding var kueri: String
    let onGantiOutlet: () -> Void
    let onBukaProfil: () -> Void
    @ViewBuilder var kategori: () -> Kategori

    var body: some View {
        VStack(spacing: 8) {
            HStack {
                HStack(spacing: 8) {
                    LogoApi()
                    JudulMerek(judul: "MENU & KATALOG", sub: "Pilihan Shawarma Otentik")
                }
                Spacer(minLength: 8)
                Button(action: onGantiOutlet) {
                    HStack(spacing: 4) {
                        Image(systemName: "mappin").font(.system(size: 11, weight: .bold)).foregroundStyle(Color.sukaOrange)
                        Text(namaOutlet).font(SukaFont.jakarta(11, weight: .bold)).foregroundStyle(.white).lineLimit(1)
                        Image(systemName: "chevron.down").font(.system(size: 10, weight: .bold)).foregroundStyle(.white)
                    }
                    .padding(.horizontal, 8).padding(.vertical, 4)
                    .background(Color.white.opacity(0.15), in: RoundedRectangle(cornerRadius: 10))
                    .overlay(RoundedRectangle(cornerRadius: 10).stroke(Color.white.opacity(0.25), lineWidth: 1))
                }
                .buttonStyle(.mentul())
                .accessibilityLabel("Outlet \(namaOutlet), ganti outlet")
                AvatarProfil(aksi: onBukaProfil)
            }
            .padding(.horizontal, 16)

            HStack(spacing: 8) {
                Image(systemName: "magnifyingglass").font(.system(size: 15)).foregroundStyle(Color.sukaMuted)
                TextField("", text: $kueri, prompt: Text("Cari menu favorit, shawarma, saus...").foregroundStyle(Color.sukaMuted))
                    .font(SukaFont.jakarta(13, weight: .medium))
                    .foregroundStyle(Color.sukaInk)
                    .submitLabel(.search)
                    .autocorrectionDisabled()
                if !kueri.isEmpty {
                    Button { kueri = "" } label: {
                        Image(systemName: "xmark.circle.fill").foregroundStyle(Color.sukaMuted)
                    }
                    .accessibilityLabel("Hapus pencarian")
                }
            }
            .padding(.horizontal, 12)
            .frame(height: 42)
            .kartuSuka(sudut: 12, elevasi: 2)
            .padding(.horizontal, 16)

            kategori()
        }
        .padding(.top, 8).padding(.bottom, 10)
        .dynamicTypeSize(...DynamicTypeSize.xxxLarge)
        .modifier(LatarKepala())
    }
}

/// 3. Kepala halaman biasa (detail, riwayat, profil): kembali/logo, judul, aksi kanan.
struct PageBrandHeader<AksiKanan: View>: View {
    let judul: String
    var subjudul: String?
    var onKembali: (() -> Void)?
    @ViewBuilder var aksiKanan: () -> AksiKanan

    var body: some View {
        HStack {
            if let onKembali {
                Button(action: onKembali) {
                    Image(systemName: "chevron.left")
                        .font(.system(size: 15, weight: .bold))
                        .foregroundStyle(Color.sukaCreamTeks)
                        .frame(width: 34, height: 34)
                        .background(Color.white.opacity(0.15), in: Circle())
                        .overlay(Circle().stroke(Color.white.opacity(0.25), lineWidth: 1))
                        .frame(width: 44, height: 44)
                }
                .buttonStyle(.mentul())
                .accessibilityLabel("Kembali")
            } else {
                LogoApi(ukuran: 34).frame(width: 44, height: 44)
            }

            VStack(spacing: 0) {
                Text(judul).font(SukaFont.lilita(17)).tracking(0.5).foregroundStyle(Color.sukaCreamTeks).lineLimit(1)
                    .accessibilityAddTraits(.isHeader)
                if let subjudul {
                    Text(subjudul).font(SukaFont.jakarta(11, weight: .bold)).foregroundStyle(Color.sukaOrange).lineLimit(1)
                }
            }
            .frame(maxWidth: .infinity)
            .padding(.horizontal, 8)

            aksiKanan().frame(minWidth: 44, minHeight: 44)
        }
        .padding(.horizontal, 12).padding(.vertical, 6)
        .dynamicTypeSize(...DynamicTypeSize.xxxLarge)
        .modifier(LatarKepala())
    }
}

extension PageBrandHeader where AksiKanan == EmptyView {
    init(judul: String, subjudul: String? = nil, onKembali: (() -> Void)? = nil) {
        self.init(judul: judul, subjudul: subjudul, onKembali: onKembali) { EmptyView() }
    }
}
